OptionsCreator.py

```diff
@@ -0,0 +1,611 @@
+                                         f"{getattr(option, 'display_name', name)}")
+        label_box = MDBoxLayout(orientation="horizontal")
+        label_anchor = MDAnchorLayout(anchor_x="right", anchor_y="center")
+        label_anchor.add_widget(option_label)
+        label_box.add_widget(label_anchor)
+
+        option_base.add_widget(label_box)
+        if issubclass(option, NamedRange):
+            option_base.add_widget(self.create_named_range(option, name))
+        elif issubclass(option, Range):
+            option_base.add_widget(self.create_range(option, name))
+        elif issubclass(option, Toggle):
+            option_base.add_widget(self.create_toggle(option, name))
+        elif issubclass(option, TextChoice):
+            option_base.add_widget(self.create_text_choice(option, name))
+        elif issubclass(option, Choice):
+            option_base.add_widget(self.create_choice(option, name))
+        elif issubclass(option, FreeText):
+            option_base.add_widget(self.create_free_text(option, name))
+        elif issubclass(option, OptionList):
+            option_base.add_widget(self.create_option_set_list(option, name, world))
+        elif issubclass(option, OptionSet):
+            option_base.add_widget(self.create_option_set_list(option, name, world))
+        else:
+            option_base.add_widget(MDLabel(text="This option isn't supported by the option creator.\n"
+                                                "Please edit your yaml manually to set this option."))
+
+        if option_can_be_randomized(option):
+            def randomize_option(instance: Widget, value: str):
+                value = value == "down"
+                if value:
+                    self.options[name] = "random-" + str(self.options[name])
+                else:
+                    self.options[name] = self.options[name].replace("random-", "")
+                    if self.options[name].isnumeric() or self.options[name] in ("True", "False"):
+                        self.options[name] = eval(self.options[name])
+
+                base_object = instance.parent.parent
+                label_object = instance.parent
+                for child in base_object.children:
+                    if child is not label_object:
+                        child.disabled = value
+
+            default_random = option.default == "random"
+            random_toggle = ToggleButton(MDButtonText(text="Random?"), size_hint_x=None, width=dp(100),
+                                         state="down" if default_random else "normal")
+            random_toggle.bind(state=randomize_option)
+            label_box.add_widget(random_toggle)
+            if default_random:
+                randomize_option(random_toggle, "down")
+
+        return option_base
+
+    def create_options_panel(self, world_button):
+        self.option_layout.clear_widgets()
+        self.options.clear()
+        cls: typing.Type[World] = world_button.world_cls
+
```

Comment:
```suggestion
        for old_button in self.scrollbox.layout.children:
            if old_button.children[0].text == self.current_game:
                old_button.state = "normal"
                break

```

Resets the ToggleButton highlights for the game list, before `self.current_game` is updated.

Reply:
Or better yet, skip the for loop, just cache the currently selected button.

```py
    def tap_world_button(self, world_button: ToggleButton):
        if self.current_world_button is not None:
            self.current_world_button.state = "normal"
        if self.current_world_button == world_button:
            self.current_world_button.state = "down"
        self.current_world_button = world_button

        self.create_options_panel(world_button)
```